### ISSUE TITLE
Unable to take screenshots (as well as other actions) with browser in headless:true 

### ISSUE THREAD

**Issue by mmarsella** [LucianoGanga/simple-headless-chrome#53]
Unable to take screenshots when setting the browser to headless.  I noticed that the window's outerHeight is also 0 when headless is turned on (maybe this is intended and I misunderstood) but some expected behavior while developing with headless:false is altered when headless:true.  Here is an example:  

```
const HeadlessChrome = require('simple-headless-chrome')
const browser = new HeadlessChrome({
  headless: true 
})

async function test(){

	await browser.init()

	// Open up a new tab
	let mainTab = await browser.newTab();

	// Log what is output in the 'browser' console
	mainTab.onConsole(async function(e){
	  console.log(`Console Says: `, e[0].value)
	})

	//Go to any site
	await mainTab.goTo('https://www.google.com');

	// Resize Screen 
	await mainTab.resizeFullScreen()

	// Log what the window height is in the console
	await mainTab.evaluate(getWindowHeight, null); 
	
	// Take a screenshot
	await mainTab.saveScreenshot('./screenshots/test.png');

	browser.close();
}

function getWindowHeight(){
	console.log("WINDOW ====> " + window.outerHeight);
}
test();
```

In the console I receive:  `Console Says:  WINDOW ====> 0`

In this case, i get a screenshot of nothing.

When I set headless to false, I receive this output in the console:

`Console Says:  WINDOW ====> 1009`

I receive a screenshot of google.com in this case.

Hope the above example helps and let me know if I can provide other insight.  Like I mentioned above, taking screenshots is one of the few features that are breaking when headless is turned on (due to the lack of a window maybe? not sure....).

Thanks 

M

**Comment by LucianoGanga** [comment]
Well! I finally got some time to do this. 

I made a few changes in the way the screenshots are taken, because the Chrome DevTools Protocol deprecated some methods. 

So I finally came with this example: 
https://github.com/LucianoGanga/simple-headless-chrome/blob/master/examples/screenCapture.js

Where you can take screenshots in headless mode again, and besides that it receives a parameter "fullPage" (to take full page screens, as you can imagine) or "selector", where you get a screenshot of the area of a selector you pass. 

I hope this helps you with the issue, and please let me know.

Everything is in v4.3.0 of this module.

I'll close the issue, but please feel free to reopen it if something doesn't work :) 

Cheers!
Lucho

**Comment by mmarsella** [comment]
Hey Luciano,

Thanks so much for getting back to me. I tried running my example with the newest version and also tried replicating your example.  Both of which threw this error:

`ERROR! { Error: Invalid parameters (fitWindow: boolean value expected)`

**Comment by LucianoGanga** [comment]
Hi!

I fixed it in v4.3.1, are you trying with that version ?



_____________________________

**Comment by mmarsella** [comment]
Yup!

**Comment by mmarsella** [comment]
Also related to this issue - while I was trying to debug this problem, I tried passing the flag:

`'--window-size=1280,1696'` to the browser instance, but it didn't seem to change anything while running headless (in regards to window visibility).

```
const browser = new HeadlessChrome({
  headless: true,
  chrome: {
    flags: [
      '--window-size=1280,1696'
    ]
  }
})
```

**Comment by LucianoGanga** [comment]
Sorry, I'm unable to see it now :(
I'm not at home. But, what chrome version are you using?

About the screen size, check the file Browser.js, you have a setting called deviceMetrics, where you can pass the width and height.

I don't know how much more time I'll be able to answer today, but I'll try.



________________________________

**Comment by mmarsella** [comment]
I am using chrome version 59.0.3071.115.

I'll take a look at Browser.js in the meantime - thanks for your time Luciano

**Comment by LucianoGanga** [comment]
Hi! I can't test with that chrome version as I have the v60.0.3112.90

About the issue: 
```js
ERROR! { Error: Invalid parameters (fitWindow: boolean value expected)
```
When I released v4.3.0 I removed the property `fitWindow` because it's deprecated in the last version of the protocol, but when I ran the tests on Travis CI, with the Stable version of Chrome, it gave me that error. So I fixed adding back that property in the options, and the tests passed. 

If I were you, I'd double check that the installed version of the module is 4.3.**1**, just in case. 

About the size. Did you have some luck? 

Tip: You can always run the code with the env variable `DEBUG="HeadlessChrome:*"` to see the module logs. 

Cheers!
Lucho

**Comment by fusionx1** [comment]
Im getting the same error.

 ```fusionx1:examples fusionx1$ node screenCapture.js 
ERROR! { Error: Invalid parameters (fitWindow: boolean value expected)
    at /Users/fusionx1/Projects/pantheon/automation/node_modules/chrome-remote-interface/lib/chrome.js:90:30
    at Chrome.handleMessage (/Users/fusionx1/Projects/pantheon/automation/node_modules/chrome-remote-interface/lib/chrome.js:289:13)
    at WebSocket.<anonymous> (/Users/fusionx1/Projects/pantheon/automation/node_modules/chrome-remote-interface/lib/chrome.js:266:27)
    at emitTwo (events.js:106:13)
    at WebSocket.emit (events.js:194:7)
    at Receiver._receiver.onmessage (/Users/fusionx1/Projects/pantheon/automation/node_modules/ws/lib/WebSocket.js:143:54)
    at Receiver.dataMessage (/Users/fusionx1/Projects/pantheon/automation/node_modules/ws/lib/Receiver.js:385:14)
    at extension.decompress (/Users/fusionx1/Projects/pantheon/automation/node_modules/ws/lib/Receiver.js:354:40)
    at _inflate.flush (/Users/fusionx1/Projects/pantheon/automation/node_modules/ws/lib/PerMessageDeflate.js:279:12)
    at afterWrite (_stream_writable.js:383:3)
    at onwrite (_stream_writable.js:374:7)
    at afterTransform (_stream_transform.js:79:3)
    at TransformState.afterTransform (_stream_transform.js:54:12)
    at Zlib.callback (zlib.js:625:5)
  response: 
   { code: -32602,
     message: 'Invalid parameters',
     data: 'fitWindow: boolean value expected' } }```

**Comment by LucianoGanga** [comment]
@fusionx1 , can you submit an example? 

Make sure you have the latest version installed (and install it through NPM)

Thanks!
Lucho

**Comment by timo-klarshift** [comment]
I am running: Chromium 60.0.3112.113  and getting the same error about fitWindow

**Comment by LucianoGanga** [comment]
Done! fitWindow problem fixed in 4.3.9!

I could finally reproduce the issue and fix it. Thanks @fusionx1 for your help on Skype the other day :) 
And sorry for taking so long to fix it. 

Cheers!
Lucoh
